Services/KioskMode/classes/class.ilKioskModeService.php

```diff
@@ -19,5 +49,22 @@ public function getViewFor(\ilObject $object) {
 	 * @param	string	$object_type	needs to be a valid object type
 	 */
 	public function hasKioskMode(string $object_type) : bool {
+		list($location, $class_name) = $this->getClassLocationForType($object_type);
+		$class_path = sprintf(
+			'%s/../../../%s/class.%s.php',
+			__DIR__, $location, $class_name
+		);
+		return file_exists($class_path);
```

Comment:
I think you could just check if `class_exists` and don't need the location. The location should already be resolved via the autoloader anyway.